commitizen/cz/__init__.py

```diff
@@ -1,21 +1,39 @@
 import importlib
 import pkgutil
-from typing import Dict, Type
+import warnings
+from typing import Dict, Iterable, Type
 
 from commitizen.cz.base import BaseCommitizen
 from commitizen.cz.conventional_commits import ConventionalCommitsCz
 from commitizen.cz.customize import CustomizeCommitsCz
 from commitizen.cz.jira import JiraSmartCz
 
+
+def discover_plugins(path: Iterable[str] = None) -> Dict[str, Type[BaseCommitizen]]:
+    """Discover commitizen plugins on the path
+
+    Args:
+        path (Path, optional): If provided, 'path' should be either None or a list of paths to look for
+    modules in. If path is None, all top-level modules on sys.path.. Defaults to None.
+
+    Returns:
+        Dict[str, Type[BaseCommitizen]]: Registry with found plugins
+    """
+    plugins = {}
+    for finder, name, ispkg in pkgutil.iter_modules(path):
+        try:
+            if name.startswith("cz_"):
+                plugins[name] = importlib.import_module(name).discover_this  # type: ignore
+        except AttributeError as e:
```

Comment:
@Woile do you think we should go with the exit code method in this case as well?

Reply:
What do you mean? like a code to easily identify where this warning is coming from?
I'm not sure it's needed, the warning should point to the line in the code AFAIK 🤔

Reply:
like what we define [here](https://github.com/commitizen-tools/commitizen/blob/master/commitizen/exceptions.py#L6) so that user can decide whether to suppress the warning. but I'm ok with this PR now. we could create another one if we think that's necessary. it won't break anything if we do that in the future

Reply:
I just approved it. If you agree with not handling this at this moment, I think we're good to merge it